File: 2287-rearrange-characters-to-make-target-string/2287-rearrange-characters-to-make-target-string.py
class Solution:
    def rearrangeCharacters(self, s: str, target: str) -> int:
        maxx = []
        for i in range(len(set(target))):
            t_count = target.count(target[i]) # count the no of occurrence of letter in target string
            s_count = s.count(target[i]) # count the no of occurrence of target letter in s
            if t_count <= s_count:
                maxx.append(s_count // t_count)
            else:
                return 0
        return min(maxx)
        

File: 2287-rearrange-characters-to-make-target-string/2287-rearrange-characters-to-make-target-string.java
class Solution {
    public int rearrangeCharacters(String s, String target) {
        
        int[] freqS = new int[26];
        int[] freqTarget = new int[26];
        int count = 0;
        for(char ch : s.toCharArray()) {
            freqS[ch-'a']++;
        }
        
        for(char ch : target.toCharArray()) {
            freqTarget[ch-'a']++;
        }

        int temp = 0;
        int min = Integer.MAX_VALUE;
        for(char ch : target.toCharArray()) {
            int x = freqS[ch-'a'];
            int y = freqTarget[ch-'a'];
            if(x >= y) {
                min = Math.min(min, x/y);
                temp++;
            }
            else{
                break;
            }
        }
        return temp==target.length() ? min : 0;
    }
}